src/main/java/com/brsanthu/googleanalytics/GoogleAnalyticsParameter.java

```diff
@@ -97,6 +100,39 @@
 	ITEM_CATEGORY("iv", new String[] {"item"}, 500),
 	CURRENCY_CODE("cu", new String[] {"transaction", "item"}, 10),
 
+	// Enhanced E-Commerce
+	PRODUCT_SKU("pr#id", new String[] {"pageview", "event"}, 500),
+	PRODUCT_NAME("pr#nm", new String[] {"pageview", "event"}, 500),
+	PRODUCT_BRAND("pr#br", new String[] {"pageview", "event"}, 500),
+	PRODUCT_CATEGORY("pr#ca", new String[] {"pageview", "event"}, 500),
+	PRODUCT_VARIANT("pr#va", new String[] {"pageview", "event"}, 500),
+	PRODUCT_PRICE("pr#pr", false, "currency", new String[] {"pageview", "event"}),
+	PRODUCT_QUANTITY("pr#qt", false, "integer", new String[] {"pageview", "event"}),
+	PRODUCT_COUPON_CODE("pr#cc", new String[] {"pageview", "event"}, 500),
+	PRODUCT_POSITION("pr#ps", false, "integer", new String[] {"pageview", "event"}),
+	PRODUCT_CUSTOM_DIMENSION("pr#cd§", new String[] {"pageview", "event"}),
+	PRODUCT_CUSTOM_METRIC("pr#cm§", false, "integer", new String[] {"pageview", "event"}),
+	PRODUCT_ACTION("pa", new String[] {"pageview", "event"}),
+	TRANSACTION_COUPON_CODE("tcc", new String[] {"pageview", "event"}),
+	PRODUCT_ACTION_LIST("pal", new String[] {"pageview", "event"}),
+	CHECKOUT_STEP("cos", false, "integer", new String[] {"pageview", "event"}),
+	CHECKOUT_STEP_OPTION("pal", new String[] {"pageview", "event"}),
+	PRODUCT_IMPRESSION_LIST_NAME("pr$nm", new String[] {"pageview", "event"}),
+	PRODUCT_IMPRESSION_SKU("pr$pi#id", new String[] {"pageview", "event"}),
```

Comment:
according to https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#il_pi_nm all these are il$pi#xx